temba/api/models.py

```diff
@@ -311,7 +311,7 @@ def trigger_flow_event(cls, run, webhook_url, node_uuid, msg, action='POST', res
                 body = 'Skipped actual send'
                 status_code = 200
 
-            if 200 <= status_code < 300:
+            if (200 <= status_code < 300) or (400 <= status_code < 500):
```

Comment:
I don't think this is right, you are now making 4** drop into the success case instead of the else as they were before.

Can either remove the else entirely and keep the attempt for json parsing in this block the same then set webhook_event.status to `COMPLETE` or `FAILED` based on the status (and throw appropriately) or repeat the attempt to parse the json and update the run in the else case. Former is probably a bit cleaner.